var/spack/repos/builtin/packages/dla-future/package.py

```diff
@@ -79,6 +79,22 @@ class DlaFuture(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("rocsolver", when="+rocm")
     depends_on("rocthrust", when="+rocm")
 
+    # nvcc 11.2 and older is unable to detect fmt::formatter specializations.
+    # DLA-Future 0.3.1 includes a workaround to avoid including fmt in device
+    # code:
+    # https://github.com/pika-org/pika/issues/870
+    # https://github.com/eth-cscs/DLA-Future/pull/1045
+    conflicts("^fmt@10:", when="@:0.3.0 +cuda ^cuda@:11.2")
+
+    # Compilation problem triggered by the bundled fmt in Umpire together with
+    # fmt 10, which only happens with GCC 9 and nvcc 11.2 and older:
+    # https://github.com/eth-cscs/DLA-Future/issues/1044
+    conflicts("^fmt@10:", when="@:0.3.0 %gcc@9 +cuda ^cuda@:11.2 ^umpire@2022.10:")
```

Comment:
I added this as a separate conflict since it describes a different compilation problem to the conflict above. However, this is a strict subset of the conflict above so the first would be sufficient. I don't know if you have a preference for keeping it separate or merging them?

Reply:
I'm fine with keeping them separate as they are better documented.